File: TreeDepth/TreeDepth.py
## Solution 1:
class Solution:
    def TreeDepth(self, root):
        # write code here
        if root is None:
            return 0
        left = self.TreeDepth(root.left)
        right = self.TreeDepth(root.right)
        return max(left, right)+1
## Solution 2:
class Solution:
    def TreeDepth(self, root):
        # write code here
        depth = 0
        if root is None:
            return depth
        q = []
        q.append(root)
        while q:
	    # count 是每一层的节点数
            count = len(q)
            while count > 0:
                node = q.pop(0)
                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)
                count-=1
	    # 遍历完一层后，深度+1
            depth+=1
        return depth

File: TreeDepth/TreeDepth.cpp
// Solution 1:
class Solution {
public:
    int TreeDepth(TreeNode* root)
    {
        if (root == NULL)
            return 0;
        int left = TreeDepth(root->left);
        int right = TreeDepth(root->right);
        return max(left, right) + 1;
    }
};
// Solution 2:
class Solution {
public:
    int TreeDepth(TreeNode* root)
    {
        int depth = 0;
        if (root == NULL)
            return depth;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty())
        {
            int count = q.size();
            while(count>0)
            {
                TreeNode* node = q.front();
                q.pop();
                if (node->left)
                    q.push(node->left);
                if (node->right)
                    q.push(node->right);
                count--;
            }
            depth++;
        }
        return depth;
    }
};
